dataset_generation/hi_wikisplit/Scraper.py

```diff
@@ -0,0 +1,174 @@
+
+        try:
+            textdata = data["compare"]["body"] \
+                            .replace('<ins class="diffchange diffchange-inline">', '') \
+                            .replace('*', '') \
+                            .replace('<ins>', '') \
+                            .replace('</ins>','') \
+                            .replace('<del class="diffchange diffchange-inline">', '') \
+                            .replace('</del>', '') 
+
+            added = textdata.split('<td class="diff-addedline">')[1] \
+                            .split("</div>")[0] \
+                            .replace('<div>', '')
+
+            deleted = textdata.split('<td class="diff-deletedline">')[1] \
+                            .split("</div>")[0] \
+                            .replace('<div>', '')
+
+            logging.info("[DIFF] successfully done on {1} and {0}".format(to_rev_id, from_rev_id))
+            return (added, deleted)
+
+        except Exception as e:
+            logging.info("[FAILED] could not get diff for revision IDs {1} and {0}"
+                    .format(to_rev_id, from_rev_id))
+            return -1
+
+    def write_raw_data_chunk(self, from_line):
+        """
+        writes a chunk of data, from the specified index
+        """
+        to_line = from_line + self.titles_per_chunk_
+        
+        output_file_path = os.path.join(self.output_dir_, "raw_data_{0}_{1}.txt".format(from_line, to_line))
+        output_file = open(output_file_path, 'wb')
+
+        for title in self.titles_[from_line:to_line]:
+            try:
+                revision_ids = self.get_revision_ids(title.strip())
+                print(revision_ids)
+                for rev_id_pair in revision_ids:
+                    print(rev_id_pair)
+                    raw_data = self.get_diff(rev_id_pair['revid'], rev_id_pair['parentid'])
+                    
+                    if(raw_data != -1):
+                        output_file.write("{0} <::::> {1} \n".format(raw_data[0], raw_data[1]).encode())
+
+            except Exception as e:
+                logging.info("[FAILED] could not get revision IDs for {0} : {1}"
+                    .format(title, e))
+                
+        output_file.close()
+        logging.info("[WROTE] data for {0} - {1} to {2}".format(from_line, to_line, output_file_path))
+            
+    def write_raw_data(self):
+        """
+        starts scraping, and writes the data chunks to the output directory
+        """
+        logging.info("[STARTED] collected data will be written to {0}\n[FOUND] {1} titles"
+                        .format(self.output_dir_, len(self.titles_)))
+
+        for i in range(self.start_from_, len(self.titles_), self.titles_per_chunk_):
+            self.write_raw_data_chunk(i)
+    
+        logging.info("[DONE] all raw data written to {0}".format(self.output_dir_))
+
+if __name__ == '__main__':
+    scraper = Scraper("hiwiki-20200503-all-titles-in-ns-0")
```

Comment:
where is this titles_file located?

Reply:
i should have made that a command line argument. I'll do that.